x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/results/add_result_logic.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../../shared/http';
+
+import { EngineLogic } from '../../../engine';
+import { Result } from '../../../result/types';
+
+interface AddResultValues {
+  isFlyoutOpen: boolean;
+  dataLoading: boolean;
+  searchQuery: string;
+  searchResults: Result[];
+}
+
+interface AddResultActions {
+  openFlyout(): void;
+  closeFlyout(): void;
+  search(query: string): { query: string };
+  onSearch({ results }: { results: Result[] }): { results: Result[] };
+}
+
+export const AddResultLogic = kea<MakeLogicType<AddResultValues, AddResultActions>>({
```

Comment:
This new logic file is pretty simple - I'm thinking we could reuse it for the query tester flyout which is super similar, or potentially just straight up convert the AddResultFlyout into the QueryTesterFlyout (just passing custom actions)